cylc/flow/task_proxy.py

```diff
@@ -237,10 +236,7 @@ def __init__(
 
         self.local_job_file_path: Optional[str] = None
 
-        if data_mode:
```

Comment:
You'd probably want to remove from `__init__` args, and use in the data-store manager.